### PR TITLE
PLT-7701, PLT-7702, PLT-7703, PLT-8427: Implement remaining 1-1 endpoints

### PR DESCRIPTION
**Pull request by bjornkihlberg**
This PR implements
- [x] `getContractSourceById` ([issue-70](https://github.com/input-output-hk/marlowe-ts-sdk/issues/70))
- [x] `getContractSourceAdjacency` ([issue-71](https://github.com/input-output-hk/marlowe-ts-sdk/issues/71))
- [x] `getContractSourceClosure` ([issue-72](https://github.com/input-output-hk/marlowe-ts-sdk/issues/72))
- [x] `getNextStepsForContract` ([issue-75](https://github.com/input-output-hk/marlowe-ts-sdk/issues/75))

## Summary by CodeRabbit

- **New Features**
  - Added a search functionality to the app.
  - Introduced new API endpoints for retrieving contract sources and their relationships.
  - Added functionality to obtain the next steps for a contract.

- **Enhancements**
  - Improved type safety with new type guard declarations.
  - Expanded the `RestClient` interface to include new methods for enhanced contract interactions.

- **Refactor**
  - Updated import statements to include additional necessary modules and functions.
  - Refined API interfaces to align with the new endpoint functionalities.

### CODE REVIEW

File: packages/runtime/client/rest/src/contract/next/endpoint.ts [hunk 1]
```diff
@@ -10,6 +12,7 @@ import { Next } from "@marlowe.io/language-core-v1/next";
 import { stringify } from "qs";
 import { DecodingError } from "@marlowe.io/adapter/codec";
 import { posixTimeToIso8601 } from "@marlowe.io/adapter/time";
+import { partiesToStrings } from "@marlowe.io/language-core-v1/participants.js";
```
[the remark this answers is not in the document]
Comment (bjornkihlberg) [reply packages/runtime/client/rest/src/contract/next/endpoint.ts:15]: Yep, that's correct. Removing the unused endpoints.

File: packages/runtime/client/rest/src/contract/endpoints/sources.ts [hunk 2]
```diff
@@ -40,3 +40,94 @@ export const createContractSources = (axiosInstance: AxiosInstance) => {
...
+        (e) => {
+          throw formatValidationErrors(e);
+        },
+        (e) => e
+      )
+    );
+  };
+
+export interface GetContractSourceAdjacencyRequest {
+  contractSourceId: BuiltinByteString;
```
Comment (hrajchert) [inline packages/runtime/client/rest/src/contract/endpoints/sources.ts:76]: I believe a type alias might clarify the intent.
```
type ContractSourceId = string 
// or 
type ContractSourceId = BuiltinByteString
```

We could asses also if a Branded type makes sense to show that not all strings are the same

File: packages/runtime/client/rest/src/contract/endpoints/sources.ts [hunk 3]
```diff
@@ -40,3 +40,94 @@ export const createContractSources = (axiosInstance: AxiosInstance) => {
...
+}
+
+export interface GetContractSourceAdjacencyResponse {
+  results: BuiltinByteString[];
```
Comment (hrajchert) [inline packages/runtime/client/rest/src/contract/endpoints/sources.ts:80]: This shows the importance of the type alias

**Review by hrajchert — CHANGES_REQUESTED** [review]
How are you testing these endpoints, or more important how can I test them? 

I don't mean the integration testing that should happen later, I'm referring to manual testing. I don't see any change to the `rest-client-flow` example with the addition of these endpoints.

Take a look at [this PR](https://github.com/input-output-hk/marlowe-ts-sdk/pull/101), where I add instructions on how to manually test the `createContractSources` endpoint

File: packages/runtime/client/rest/src/contract/endpoints/sources.ts (hunk 3 above)
Comment (bjornkihlberg) [reply packages/runtime/client/rest/src/contract/endpoints/sources.ts:80]: Fair point.

File: packages/runtime/client/rest/src/contract/endpoints/sources.ts (hunk 2 above)
Comment (bjornkihlberg) [reply packages/runtime/client/rest/src/contract/endpoints/sources.ts:76]: Have we figured out yet how we want to do branded types?

Comment (hrajchert) [reply packages/runtime/client/rest/src/contract/endpoints/sources.ts:76]: Yes, using t.brand from io-ts

**Review by hrajchert — APPROVED** [review]

**Pull request merged**